Neos.Media/Classes/Domain/Service/ImageService.php

```diff
@@ -112,7 +118,7 @@ public function processImage(PersistentResource $originalResource, array $adjust
 
         $resourceUri = $originalResource->createTemporaryLocalCopy();
 
-        $resultingFileExtension = $originalResource->getFileExtension();
+        $resultingFileExtension = ($format && in_array($format, self::$allowedFormats)) ? $format : $originalResource->getFileExtension();
```

Comment:
weak check could be changed (truthy & in_array strict comparison)
```suggestion
        $resultingFileExtension = ($format !== null && in_array($format, self::$allowedFormats, true)) ? $format : $originalResource->getFileExtension();
```